### ISSUE TITLE
azurerm_storage_management_policy  overwrites the existing rules

### ISSUE THREAD

**Issue by VivekDevops** [hashicorp/terraform-provider-azurerm#6935]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version
 0.12.17

### Affected Resource(s)

* `azurerm_storage_management_policy`

### Terraform Configuration Files

```hcl
data azurerm_storage_account storage_account {
  name = "${var.storage_account_name}"
  resource_group_name = "${var.resource_group_name}"
}

resource azurerm_storage_management_policy policy {
  storage_account_id = "${data.azurerm_storage_account.storage_account.id}"

  rule {
    name    = "EtcdBackupLifecycle"
    enabled = true
    filters {
      prefix_match = ["${var.bucket_name}/etcd"]
      blob_types   = ["blockBlob"]
    }
    actions {
      base_blob {
        delete_after_days_since_modification_greater_than          = 30
      }
      snapshot {
        delete_after_days_since_creation_greater_than = 30
      }
    }
  }
}
```

### Expected Behavior

This terraform apply should only create a new lifecycle policy  

### Actual Behavior

 It creates a new lifecycle policy, but also overwrites any of the existing lifecycle policy

### Steps to Reproduce

1. `terraform apply`

### References

* #0000

**Comment by lrxtom2** [comment]
@VivekDevops Hi, thanks for opening the issue. I will work on that, but I am sorry that there is no firm ETA for this.

**Comment by VivekDevops** [comment]
Thanks @lrxtom2 for looking into the issue. Just curious,  Is there any workaround that u r aware of until this is fixed.

**Comment by Arlington1985** [comment]
@VivekDevops, did you find any workaround?

**Comment by VivekDevops** [comment]
@Arlington1985, I am currently using the azure apis instead(Not using terraform for setting up the lifecycle rules)

https://management.azure.com/subscriptions/%s/resourceGroups/%s/providers/Microsoft.Storage/storageAccounts/%s/managementPolicies/default?api-version=2019-06-01

**Comment by andrea-cassioli-maersk** [comment]
Any progress on this issue?

**Comment by davidwneary** [comment]
I've come across this issue because I have 2 `azurerm_storage_management_policy` resources for the same storage account that are managed by 2 different terraform projects. I wonder if an alternative `azurerm_storage_management_policy_rule` resource would be useful?

**Comment by worldspawn** [comment]
@lrxtom2 any word on this (just ran into it)?

**Comment by magodo** [comment]
@VivekDevops Terraform is a declarative tool, which means you get what you have declared. In case of `azurerm_storage_management_policy` defined above, you have only declared one `rule`, therefore Terraform ensures you get exactly that rule. If you have more than one rules, you'll have to define them by specifying multiple `rule`s.

**Comment by magodo** [comment]
@Arlington1985 Yes, that is possible. Sorry, I overlooked that part. I'll put that to my todo list then.

**Comment by seblatre** [comment]
It's been almost 4 years that this issue has been released and still not fixed (Using AzureRM 3.85.0). It's a real issue for us as we are relying on a single storage account to share the logs of different components managed with different Terraform.
Any plan to get that fixed?

**Comment by musabmirza-amperon** [comment]
Same here 🤔

**Comment by gdanielson** [comment]
👍

**Comment by ltutar** [comment]
It is 15th of January 2025. Any updates on this?

**Comment by magodo** [comment]
Pending on https://github.com/Azure/azure-rest-api-specs/issues/21654. I've created an MSFT ticket for this as well.

**Comment by gmanera** [comment]
Any updates on this?
